src/mgmt/srv_drpc.c

```diff
@@ -2209,325 +2207,189 @@ ds_mgmt_drpc_bio_health_query(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 	int			 rc = 0;
 
 	/* Unpack the inner request from the drpc call body */
-	req = ctl__dev_identify_req__unpack(&alloc.alloc, drpc_req->body.len, drpc_req->body.data);
+	req = ctl__dev_replace_req__unpack(&alloc.alloc, drpc_req->body.len, drpc_req->body.data);
 
 	if (alloc.oom || req == NULL) {
-		drpc_resp->status = DRPC__STATUS__FAILURE;
-		D_ERROR("Failed to unpack req (dev identify)\n");
+		D_ERROR("Failed to unpack req (dev replace)\n");
+		drpc_resp->status = DRPC__STATUS__FAILED_UNMARSHAL_PAYLOAD;
 		return;
 	}
 
-	if (strlen(req->dev_uuid) == 0) {
-		D_ERROR("Device UUID for identify command is empty\n");
+	D_INFO("Received request to replace device %s with device %s\n",
+	       req->old_dev_uuid, req->new_dev_uuid);
+
+	D_ALLOC_PTR(resp);
+	if (resp == NULL) {
+		D_ERROR("Failed to allocate daos response ref\n");
 		drpc_resp->status = DRPC__STATUS__FAILURE;
-		ctl__dev_identify_req__free_unpacked(req, &alloc.alloc);
+		ctl__dev_replace_req__free_unpacked(req, &alloc.alloc);
 		return;
 	}
 
-	D_INFO("Received request to identify device %s\n", req->dev_uuid);
+	/* Response status is populated with SUCCESS on init. */
+	ctl__dev_manage_resp__init(resp);
 
-	D_ALLOC_PTR(resp);
+	D_ALLOC_PTR(resp->device);
 	if (resp == NULL) {
+		D_ERROR("Failed to allocate daos response device ref\n");
 		drpc_resp->status = DRPC__STATUS__FAILURE;
-		D_ERROR("Failed to allocate daos response ref\n");
-		ctl__dev_identify_req__free_unpacked(req, &alloc.alloc);
+		drpc_dev_manage_free(resp);
+		ctl__dev_replace_req__free_unpacked(req, &alloc.alloc);
 		return;
 	}
 
-	/* Response status is populated with SUCCESS on init. */
-	ctl__dev_identify_resp__init(resp);
+	ctl__smd_device__init(resp->device);
 	/* Init empty strings to NULL to avoid error cleanup with free */
-	resp->dev_state = NULL;
-	resp->dev_uuid = NULL;
+	resp->device->uuid = NULL;
 
-	if (uuid_parse(req->dev_uuid, dev_uuid) != 0) {
-		D_ERROR("Unable to parse device UUID %s: %d\n",
-			req->dev_uuid, rc);
-		uuid_clear(dev_uuid); /* need to set uuid = NULL */
+	if (uuid_parse(req->old_dev_uuid, old_uuid) != 0) {
+		D_ERROR("Old device UUID (%s) is invalid\n", req->old_dev_uuid);
+		D_GOTO(pack_resp, rc = -DER_INVAL);
 	}
 
-	rc = ds_mgmt_dev_identify(dev_uuid, resp);
-	if (rc != 0)
-		D_ERROR("Failed to set LED to IDENTIFY on device:%s\n",
-			req->dev_uuid);
-
-	resp->status = rc;
-	len = ctl__dev_identify_resp__get_packed_size(resp);
-	D_ALLOC(body, len);
-	if (body == NULL) {
-		drpc_resp->status = DRPC__STATUS__FAILURE;
-		D_ERROR("Failed to allocate drpc response body\n");
-	} else {
-		ctl__dev_identify_resp__pack(resp, body);
-		drpc_resp->body.len = len;
-		drpc_resp->body.data = body;
+	if (uuid_parse(req->new_dev_uuid, new_uuid) != 0) {
+		D_ERROR("New device UUID (%s) is invalid\n", req->new_dev_uuid);
+		D_GOTO(pack_resp, rc = -DER_INVAL);
 	}
 
-	ctl__dev_identify_req__free_unpacked(req, &alloc.alloc);
+	/* TODO: Implement no-reint device replacement option */
```

Comment:
Is there a ticket for this?

Reply:
This isn't something I added, I just moved it. @NiuYawei or @sydneyvanda do you have a ticket number or this? I will update in the following PR unless I have to re-push.

Reply:
Okay, no worries.

Reply:
yes this is https://daosio.atlassian.net/browse/DAOS-6283

Reply:
updated on PR https://github.com/daos-stack/daos/pull/10768